matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/crosssigning/ComputeTrustTask.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.matrix.android.internal.crypto.crosssigning
+
+import im.vector.matrix.android.api.crypto.RoomEncryptionTrustLevel
+import im.vector.matrix.android.api.extensions.orFalse
+import im.vector.matrix.android.api.session.crypto.crosssigning.MXCrossSigningInfo
+import im.vector.matrix.android.internal.crypto.store.IMXCryptoStore
+import im.vector.matrix.android.internal.task.Task
+import javax.inject.Inject
+
+internal interface ComputeTrustTask : Task<ComputeTrustTask.Params, RoomEncryptionTrustLevel> {
+    data class Params(
+            val userList: List<String>
+    )
+}
+
+internal class DefaultComputeTrustTask @Inject constructor(
+        val cryptoStore: IMXCryptoStore
+) : ComputeTrustTask {
+
+    override suspend fun execute(params: ComputeTrustTask.Params): RoomEncryptionTrustLevel {
+        val userIds = params.userList
+        val allTrusted = userIds
+                .filter { getUserCrossSigningKeys(it)?.isTrusted() == true }
+
+        val allUsersAreVerified = userIds.size == allTrusted.size
+
+        return if (allTrusted.isEmpty()) {
+            RoomEncryptionTrustLevel.Default
```

Comment:
We do not use the legacy method here?

Reply:
No. But we have iconsistency though, the room will be blackshield but the profile will be red (profile uses legacy) :/
Still moving https://docs.google.com/document/d/1RFUN0YRrOs8QZwiblaUbs4wWR0BjA0hBJ27rKJ57Zvk/edit?skip_itp2_check=true&pli=1#heading=h.jvldvebnsc7f

Traked here: https://github.com/vector-im/riotX-android/issues/963

Reply:
yes, thanks